lib/view_component/compiler.rb

```diff
@@ -31,6 +31,8 @@ def compile(raise_errors: false, force: false)
       return if compiled? && !force
       return if component_class == ViewComponent::Base
 
+      component_class.superclass.compile(raise_errors: raise_errors) if development? && templates.empty?
```

Comment:
Wouldn't we only want to do this if there is no `call` method either?

Reply:
@joelhawksley I've added a test that doesn't directly assert that we don't call `compile` on the superclass when `call` is defined on the inheriting component, but that we should expect to raise an error in the case when `compile` _is_ called on the superclass.